src/Aspire.Hosting.Azure.AIFoundry/Internal/FoundryLocalManager.cs

```diff
@@ -0,0 +1,552 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+// --------------------------------------------------------------------------------------------------------------------
+// <copyright company="Microsoft">
+//   Copyright (c) Microsoft. All rights reserved.
+// </copyright>
+// --------------------------------------------------------------------------------------------------------------------
+
+namespace Aspire.Hosting.Azure.Internal;
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Globalization;
+using System.Linq;
+using System.Net.Http.Json;
+using System.Net.Mime;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+
+[JsonConverter(typeof(JsonStringEnumConverter<DeviceType>))]
+
+internal enum DeviceType
+{
+    CPU,
+    GPU,
+    NPU,
+    Invalid
+}
+
+[JsonConverter(typeof(JsonStringEnumConverter<ExecutionProvider>))]
+internal enum ExecutionProvider
+{
+    Invalid,
+    CPUExecutionProvider,
+    WebGpuExecutionProvider,
+    CUDAExecutionProvider,
+    QNNExecutionProvider
+}
+
+internal sealed partial class FoundryLocalManager : IDisposable, IAsyncDisposable
```

Comment:
This is a lot of logic that doesn't seem to have any logging. Make me wonder if it should be a global resource that shows up in the dashboard.